src/components/Icon/Icon.css

```diff
@@ -11,3 +17,36 @@
 .y-icon__isBlock {
   display: block;
 }
+
+/**
+  * Overrides for specific font sizes for crisper lines and better vertical spacing.
+  * Relies on the application using YamUI components instead of additional CSS to specify textSize.
+  */
+.y-textSize-xxLarge .y-icon {
+  height: 2.3rem;
+  width: 2.3rem;
+  vertical-align: -0.1rem;
+}
+.y-textSize-xLarge .y-icon {
+  height: 2.1rem;
+  width: 2.1rem;
+  vertical-align: -0.1rem;
+}
+.y-textSize-large .y-icon {
+  height: 1.6rem;
+  width: 1.6rem;
+}
+/* medium matches the default values */
+.y-textSize-mediumSub .y-icon {
+  height: 1.4rem;
```

Comment:
@swese44 would you consider using fontsize variables here instead so that these change relatively?

Reply:
the icons change relative to the font-size from the nearest parent `<Text size={...}>` or `<Block textSize={...}>`. at a very local level this lets us use specific rem values to ensure crisp lines (instead of using em or %).

at a higher design level, we have a finite list of font-size + line-height combinations we support which match our vertical rhythm, we want people using YamUI components instead of writing their own CSS.